app/assets/javascripts/discourse/app/components/d-modal.js

```diff
@@ -141,6 +141,10 @@ export default Component.extend({
         focusTarget = this.element.querySelector(
           ".modal-body input, .modal-body button, .modal-footer input, .modal-footer button"
         );
+
+        if (!focusTarget) {
+          focusTarget = this.element.querySelector(".modal-header button");
```

Comment:
Why not add to the selector above? it's already doing multiple things in one line.